ports/stm32/rfcore.c

```diff
@@ -605,4 +618,67 @@ void IPCC_C1_RX_IRQHandler(void) {
     IRQ_EXIT(IPCC_C1_RX_IRQn);
 }
 
+// AN5185
+#define MAGIC_FUS_ACTIVE 0xA94656B9
+// AN5289
+#define MAGIC_IPCC_MEM_INCORRECT 0x3DE96F61
+
+STATIC mp_obj_t rfcore_fus_active(void) {
+    if (ipcc_mem_dev_info_tab.fus.table_state == MAGIC_IPCC_MEM_INCORRECT) {
+        mp_raise_OSError(MP_EINVAL);
+    }
+    return mp_obj_new_bool(ipcc_mem_dev_info_tab.fus.table_state == MAGIC_FUS_ACTIVE);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(rfcore_fus_active_obj, rfcore_fus_active);
+
+STATIC mp_obj_t get_version_tuple(uint32_t data) {
+    mp_obj_t items[] = {
+        MP_OBJ_NEW_SMALL_INT(data >> 24), MP_OBJ_NEW_SMALL_INT(data >> 16 & 0xFF), MP_OBJ_NEW_SMALL_INT(data >> 8 & 0xFF), MP_OBJ_NEW_SMALL_INT(data >> 4 & 0xF), MP_OBJ_NEW_SMALL_INT(data & 0xF)
+    };
+    return mp_obj_new_tuple(5, items);
+}
+
+STATIC mp_obj_t rfcore_fus_version(void) {
+    uint32_t v = (rfcore_fus_active() == mp_const_true) ? ipcc_mem_dev_info_tab.fus.fus_version : ipcc_mem_dev_info_tab.ws.fus_version;
+    return get_version_tuple(v);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(rfcore_fus_version_obj, rfcore_fus_version);
+
+STATIC mp_obj_t rfcore_ws_version(void) {
+    uint32_t v = (rfcore_fus_active() == mp_const_true) ? ipcc_mem_dev_info_tab.fus.ws_version : ipcc_mem_dev_info_tab.ws.fw_version;
+    return get_version_tuple(v);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(rfcore_ws_version_obj, rfcore_ws_version);
```

Comment:
These version functions could be combined into `rfcore.version(id)`, where id=0/1 is a constant specifying FUS or WS... then maybe also `rfcore.active(id)`...?  this API here doesn't need to be pretty, and it'd be fair enough to say it's private at the moment.

Reply:
Done.

stm.rfcore_status() returns the first word of the device info table (and the Python knows how to handle the differnet values)
stm.rfcore_fw_version(id) takes 0 or 1 for FUS or WS.
stm.rfcore_sys_hci (as before)